Source/UINavigation/Private/UINavHorizontalComponent.cpp

```diff
@@ -11,6 +12,8 @@ void UUINavHorizontalComponent::NavigateLeft()
 	OnUpdated();
 	ParentWidget->OnHorizCompNavigateLeft(NavButton->ButtonIndex);
 	ParentWidget->OnHorizCompUpdated(NavButton->ButtonIndex);
+
+	if(ParentWrapper != nullptr) ParentWrapper->NavComponentUpdated(); 
```

Comment:
I'd rather you do this differently.
Instead of doing all this, you should create a new function in the UINavWidget (e.g. NotifyOnHorizCompNavigated/Updated) and there call OnHorizCompNavigateLeft and also check if the horiz component that was updated belongs to one of the UINavCollections (check CollectionNavigateTo in UINavWidget for reference) and call the OnHorizCompUpdated for each UINavCollection there.
With this approach, the only changes from this PR that would be kept would be adding the OnNavComponentUpdated event in the UINavCollection.